src/tests/mod.rs

```diff
@@ -17,3 +19,63 @@ macro_rules! test_amber {
 pub fn compile_code<T: Into<String>>(code: T) -> String {
     AmberCompiler::new(code.into(), None).compile().unwrap().1
 }
+
+/// Compile code and compare it with the expected value.
+/// If any difference is found panic and show a friendly diff
+pub fn comp(source: &str, expected: &str) {
+    let mut exp = String::with_capacity(500);
+    let exp = {
+        // Dedent code blocks based on the first line after `r#"`
+        let mut lines = expected.lines();
+        lines.next();
+        let line = lines.next().unwrap_or("");
+        let indent = line.chars().take_while(|&c| c.is_whitespace()).count();
+        exp.push_str(&line[indent..]);
+        exp.push('\n');
+        for line in lines {
+            if line.len() >= indent {
+                exp.push_str(&line[indent..]);
+                exp.push('\n');
+            } else {
+                panic!("Incorrect indentation on line <<<{line}>>>.\nExpected {indent} characters")
+            }
+        }
+        exp.trim_end()
+    };
+
+    let compiled = compile_code(source);
+
+    let header = r#"#!/usr/bin/env bash
```

Comment:
you should load the header from its file

Reply:
In a unit test the goal is to minimize the complexity of the validation and ensure that the output are manually verified - this is why the header string is hardcoded. [Ideally the trimming of newlines should also be removed in future to make tests simpler and more strict.]
Anyhow if you prefer I can switch to reading the header from file anyways.

Reply:
that simply breaks DRY. what if we change the header? the test will break as well

so its gotta be `include_str`'d

Reply:
OK, I'll update it. (For the record, breaking DRY and having to manually update unit tests is part of the point of unit testing)